tests/manage/z_cluster/nodes/test_automated_recovery_from_failed_nodes_reactive_ms.py

```diff
@@ -151,64 +159,58 @@ def check_automated_recovery_from_terminated_node(nodes):
-    3) The new OSD pods with the same ids that come up, should be in a Pending state.
-    4) Schedule the worker node.
-    5) The OSD pods associated with the node, should back into a Running state, and come up
-        on the same node.
+    3) Schedule the worker node.
+    4) The OSD pods associated with the node, should back into a Running state, and come up
+       on the same node or a different node in the same zone.
 
     """
     osd_node_name = random.choice(get_osd_running_nodes())
     old_osd_pod_ids = get_node_osd_ids(osd_node_name)
     log.info(f"osd pod ids: {old_osd_pod_ids}")
     node_osd_pods = get_osd_pods_having_ids(old_osd_pod_ids)
 
-    osd_labelled_nodes = set(machine.get_labeled_nodes(constants.OSD_NODE_LABEL))
-    osd_running_nodes = set(get_osd_running_nodes())
-
     unschedule_nodes([osd_node_name])
     log.info(f"Successfully unschedule the node: {osd_node_name}")
 
     log.info("Delete the node osd pods")
     delete_pods(node_osd_pods)
 
-    new_osd_pods = wait_for_osd_pods_having_ids(osd_ids=old_osd_pod_ids)
-    new_osd_pod_names = [p.name for p in new_osd_pods]
-
-    wnodes = get_worker_nodes()
-    if len(wnodes) <= 3 or osd_labelled_nodes == osd_running_nodes:
-        expected_pods_status = constants.STATUS_PENDING
-    else:
-        expected_pods_status = constants.STATUS_RUNNING
-
-    log.info(
-        f"Verify the new osd pods {new_osd_pod_names} go into a {expected_pods_status} state"
-    )
-    res = wait_for_pods_to_be_in_statuses(
-        [expected_pods_status],
-        new_osd_pod_names,
-        raise_pod_not_found_error=True,
-    )
-    assert res, f"Not all the node osd pods are in a {expected_pods_status} state"
-
     log.info(f"Wait for the node: {osd_node_name} to be scheduled")
     schedule_nodes([osd_node_name])
     log.info(f"Successfully scheduled the node {osd_node_name}")
 
-    if len(wnodes) <= 3 or osd_labelled_nodes == osd_running_nodes:
-        assert wait_for_osd_ids_come_up_on_node(osd_node_name, old_osd_pod_ids)
-        log.info(
-            f"the osd ids {old_osd_pod_ids} Successfully come up on the node {osd_node_name}"
-        )
+    # Wait for correct OSD distribution. This will also wait for the necessary number of OSD pods to be Running
+    retry((AssertionError), tries=10, delay=60)(verify_osd_distribution_on_provider)
+
+    # Verify that OSD pods are running on the node
+    osd_ids_on_node = get_node_osd_ids(osd_node_name)
+    assert osd_ids_on_node, f"No OSD pod is running on the node {osd_node_name}."
+    log.info(f"OSD {osd_ids_on_node} are running on the node {osd_node_name}.")
+
+    # The OSD IDs that were present on the node should be in running state now
+    osds_with_old_id = get_osd_pods_having_ids(old_osd_pod_ids)
+    assert len(set(osds_with_old_id)) == len(
+        set(old_osd_pod_ids)
+    ), f"One or more of the OSD IDs {old_osd_pod_ids} which were running on the node are not present now"
```

Comment:
Same here.

Reply:
Wait is not needed here due to the reason described above.